src/Native/System.Native/pal_io.cpp

```diff
@@ -793,3 +794,17 @@ extern "C" int32_t Write(int32_t fd, const void* buffer, int32_t bufferSize)
     assert(count >= -1 && count <= bufferSize);
     return static_cast<int32_t>(count);
 }
+
+extern "C" int32_t GetWindowSize(WinSize* windowSize)
+{
+    assert(windowSize != nullptr);
+
+    int error = ioctl(1/*STDOUT_FILENO*/, TIOCGWINSZ, windowSize);
```

Comment:
> STDOUT_FILENO should be defined in unistd.h; it's not working?

I just checked it works :). Taken care.

> Do we need to do cmake feature detection for ioctl and TIOCGWINSZ and then make this conditional on their availability, otherwise setting errno to ENOTSUP and returning -1?

I am not sure how to do that. Could you please point me in the right direction?